src/app/components/RoundedBalance/__tests__/index.test.tsx

```diff
@@ -1,41 +1,66 @@
-import { render, screen } from '@testing-library/react'
+import { Matcher, render, screen } from '@testing-library/react'
 import { RoundedBalance } from './../'
 
+/**
+ * Extract the text content from an element.
+ *
+ * Will only work if the element contains only a series of text elemeents.
+ */
+const extractTextFromElement = (element: Element): string =>
+  element.textContent +
+  Array.from(element.childNodes)
+    .map(c => c.textContent ?? '')
+    .join('')
+
+const hasGrandChildren = (element: Element): boolean =>
+  Array.from(element.childNodes).some(node => node.hasChildNodes())
+
+const hasOnlyOneLevelOfChildren = (element: Element): boolean => !hasGrandChildren(element)
+
+/**
+ * Get a text match function that is capable for searching for a given pattern,
+ * even if the pattern is spread to multiple text nodes.
+ */
+const getFlexMatcherFor =
+  (pattern: string): Matcher =>
+  (content, element) =>
+    !!element && extractTextFromElement(element).startsWith(content) && hasOnlyOneLevelOfChildren(element)
+
 describe('RoundedBalance', () => {
   it('should keep original value', () => {
     render(<RoundedBalance value="0.00075" ticker="USDC" />)
 
-    expect(screen.getByText('0.00075 USDC')).toBeInTheDocument()
+    expect(screen.getByText(getFlexMatcherFor('0.00075 USDC'))).toBeInTheDocument()
```

Comment:
this doesn't work.

Reply:
OK, text matching fixed. Fixing the tests revealed a problem, which is now also fixed.

Reply:
Works. But I think this is how it natively works too. Is this needed?

Reply:
When you have two text elements side by side, like this: 
![image](https://github.com/oasisprotocol/explorer/assets/2093792/4031fc94-5eed-4fec-838e-1354899ff7d6)

... and you search for the concatenated text, no match is found. There is an error message that explicitly explains that this might be because the text is divided into multiple elements, and in this case you should provide a more flexible matcher function. 

I think they could just as well provide it themselves, but apparently they don't, or at lest I didn't find it. The default matcher was definitely inadequate.

Reply:
If I remove getFlexMatcherFor tests still pass

Reply:
OK I have no idea what happened here, then. For they definitely didn't pass before. Maybe it was an independent problem which has been resolved since, and the crafty error message mislead me about the reason? I don't know.

In any case I have no removed the unnecessary changes.